libs/libc/stdio/lib_vasprintf.c

```diff
@@ -58,7 +58,7 @@
 #define ap1 ap
 #define ap2 ap
 
-#if defined(CONFIG_ARCH_X86)
+#if defined(CONFIG_ARCH_X86) || defined(CONFIG_ARCH_X86_64)
```

Comment:
Should we change like vsyslog to avoid the same problem happen on the new arch:
#ifdef va_copy
      va_list copy;

      va_copy(copy, ap);
      nx_vsyslog(priority, fmt, &copy);
      va_end(copy);
#else
      nx_vsyslog(priority, fmt, &ap);
#endif

Reply:
I think it is OK.  If anything this is a place where we should add the conditional to avoid the copy if the arch is not leaving it on the stack.

From `stdarg.h` in gcc 
```
#if !defined(__STRICT_ANSI__) || __STDC_VERSION__ + 0 >= 199900L || defined(__GXX_EXPERIMENTAL_CXX0X__)
#define va_copy(d,s)        __builtin_va_copy(d,s)
#endif
#define __va_copy(d,s)        __builtin_va_copy(d,s)
```

and `stdarg.h` in clang
```
/* GCC always defines __va_copy, but does not define va_copy unless in c99 mode
 * or -ansi is not specified, since it was not part of C90.
 */
#define __va_copy(d,s) __builtin_va_copy(d,s)

#if __STDC_VERSION__ >= 199901L || __cplusplus >= 201103L || !defined(__STRICT_ANSI__)
#define va_copy(dest, src)  __builtin_va_copy(dest, src)
#endif
```

Reply:
We do need to do something differently.  Architectural dependencies or conditional logic  are not permitted in common code.  We control all architecture-specific logic under libs/linc/machine or in header files at include/arch.

Reply:
Is there a downside to just always calling the copy as is done with vsyslog? I'm not sure how to avoid the stack usage without having machine specific code in the function.

Reply:
It isn't a big issue because va_copy is a simple pointer assignment for no x64 like platform.